mds/apis/prv_api/service_areas.py

```diff
@@ -81,6 +95,36 @@ def create(self, *args, **kwargs):
     def update(self, *args, **kwargs):
         return super()._update(*args, **kwargs)
 
+    @action(methods=["post"], url_path="import", detail=False)
+    def import_polygons(self, request, pk=None):
+        polygons = request.data.get("polygons", None)
+
+        if not isinstance(polygons, list):
+            return Response(status=400)
+
+        for polygon in polygons:
+            geom = polygon.get("geom", None)
+
+            if geom and geom["type"] == "Polygon":
+                try:
+                    areas = []
+                    for area_label in polygon.get("areas", []):
+                        defaults = {
+                            "color": "#%06x" % random.randint(0, 0xFFFFFF)
+                        }
+                        # Create new Area if doesn't exist (based on label)
+                        area = models.Area.objects.get_or_create(
+                            label=area_label, defaults=defaults
+                        )[0]
+                        areas.append(area)
+                    poly = models.Polygon.objects.create(
```

Comment:
This for loop could generate a lot of db queries for large imports.
You should consider batching them (by for example creating models.Polygon instances and then calling https://docs.djangoproject.com/fr/2.1/ref/models/querysets/#bulk-create .
You should maybe prefetch and batch areas too.